data_structures/linked_list/singly_link_list_deletion.c

```diff
@@ -1,98 +1,112 @@
-/*Includes structure for a node which can be use to make new nodes of the Linked List.
-  It is assumed that the data in nodes will be an integer, though
-  function can be modified according to the data type, easily.
-  deleteNode deletes a node when passed with a key of the node.
+/* 
+ *Includes structure for a node which can be use to make new nodes of the Linked List. 
+ *It is assumed that the data in nodes will be an integer, though 
+ *function can be modified according to the data type, easily. 
+ *deleteNode deletes a node when passed with a key of the node.
 */
+
 #include<stdio.h>
+
 struct node
-{int info;
- struct node *link;
+{
+	int info;
+	struct node *link;
 };
-struct node *start=NULL;
-///////////////////////////////////////////////////////////
-struct node * createnode()//function to create node
+
+struct node *start = NULL;
+
+struct node *createnode()
 {
-  struct node *t;
-  t=(struct node*)malloc(sizeof(struct node));
-  return(t);
+	struct node *new;
+	new = (struct node*)malloc(sizeof(struct node));
+	if (new == NULL) {
+		new = NULL;
+		goto out_failure;
+	}
+	
+out_success:
+	return new;
+out_failure:
+	return new;
```

Comment:
what's the point of using 2 goto labels if both are doing the same thing _(return new)_ ?